extensions/sql-migration/src/models/stateMachine.ts

```diff
@@ -434,6 +440,127 @@ export class MigrationStateModel implements Model, vscode.Disposable {
 		return this._skuRecommendationResults;
 	}
 
+
+	public async getSourceConnectionString(): Promise<string> {
+		return await azdata.connection.getConnectionString(this._sourceConnectionId, true);
+	}
+
+	public async setTargetServerName(): Promise<void> {
+		// If target server name has already been set, we can skip this part
+		if (this._targetServerName) {
+			return;
+		}
+
+		if (isSqlManagedInstance(this._targetServerInstance) || isAzureSqlDatabaseServer(this._targetServerInstance)) {
+			this._targetServerName = this._targetServerName ?? this._targetServerInstance.properties.fullyQualifiedDomainName;
+		}
+	}
+
+	public async getTargetConnectionString(): Promise<string> {
+		await this.setTargetServerName();
+		const connectionProfile = getConnectionProfile(
+			this._targetServerName,
+			this._targetServerInstance.id,
+			this._targetUserName,
+			this._targetPassword);
+
+		const result = await azdata.connection.connect(connectionProfile, false, false);
+		if (result.connected && result.connectionId) {
+			return azdata.connection.getConnectionString(result.connectionId, true);
+		}
+
+		return '';
+	}
+
+	private updateLoginMigrationResults(newResult: mssql.StartLoginMigrationResult): void {
+		if (this._loginMigrationsResult && this._loginMigrationsResult.exceptionMap) {
+			for (var key in newResult.exceptionMap) {
+				this._loginMigrationsResult.exceptionMap[key] = [...this._loginMigrationsResult.exceptionMap[key] || [], newResult.exceptionMap[key]]
+			}
+		} else {
+			this._loginMigrationsResult = newResult;
+		}
+	}
+
+	public async migrateLogins(): Promise<Boolean> {
+		try {
+			const sourceConnectionString = await this.getSourceConnectionString();
+			const targetConnectionString = await this.getTargetConnectionString();
+			console.log('Starting Login Migration at: ', new Date());
+
+			console.time("migrateLogins")
+			var response = (await this.migrationService.migrateLogins(
+				sourceConnectionString,
+				targetConnectionString,
+				this._loginsForMigration.map(row => row.loginName),
+				this._aadDomainName
+			))!;
+			console.timeEnd("migrateLogins")
+
+			this.updateLoginMigrationResults(response)
+		} catch (error) {
+			console.log('Failed Login Migration at: ', new Date());
```

Comment:
console.log calls can be removed where telemetery, logError calls are made as they also call console.log

Reply:
i will remove this before checking in. I have them in for debugging purposes right now (just so I know what to search for through in console for errors)